OpenRA.Game/Game.cs

```diff
@@ -585,7 +585,7 @@ static void InnerLogicTick(OrderManager orderManager)
 
 					var isNetTick = LocalTick % NetTickScale == 0;
 
-					if (!isNetTick || orderManager.IsReadyForNextFrame)
```

Comment:
Checking this before ticking the order manager guaranteed that we had all the orders for this frame before we tried to send them. With the new approach we will send any orders we have by the first time we check, and then (i think?) if `SendNetFrameOrdersAndCheckReady` returns false the first time any further orders will be delayed until the next tick. Isn't this just adding an additional tick of latency?

Reply:
Orders that come in during a stall will be delayed until the next frame from when they would be in the current version, yes. I could fix this by checking if we are blocked on any other clients first, and only sending the orders now if we aren't. This would only take effect when order latency is 0, since when it is above one we are always ready if every other player is ready.